server-common/src/main/java/org/apache/kafka/server/util/CommandLineUtils.java

```diff
@@ -0,0 +1,201 @@
+        }
+        if (isPrintVersionNeeded(commandOpts)) {
+            printVersionAndDie();
+        }
+    }
+
+    /**
+     * Check that all the listed options are present.
+     */
+    public static void checkRequiredArgs(OptionParser parser, OptionSet options, OptionSpec<?>... requiredList) {
+        for (OptionSpec<?> arg : requiredList) {
+            if (!options.has(arg)) {
+                printUsageAndDie(parser, String.format("Missing required argument \"%s\"", arg));
+            }
+        }
+    }
+
+    /**
+     * Check that none of the listed options are present.
+     */
+    public static void checkInvalidArgs(OptionParser parser,
+                                        OptionSet options,
+                                        OptionSpec<?> usedOption,
+                                        OptionSpec<?>... invalidOptions) {
+        if (options.has(usedOption)) {
+            for (OptionSpec<?> arg : invalidOptions) {
+                if (options.has(arg)) {
+                    printUsageAndDie(parser, String.format("Option \"%s\" can't be used with option \"%s\"", usedOption, arg));
+                }
+            }
+        }
+    }
+
+    /**
+     * Check that none of the listed options are present.
+     */
+    public static void checkInvalidArgs(OptionParser parser,
+                                        OptionSet options,
+                                        OptionSpec<?> usedOption,
+                                        Set<OptionSpec<?>> invalidOptions) {
+        OptionSpec<?>[] array = new OptionSpec<?>[invalidOptions.size()];
+        invalidOptions.toArray(array);
+        checkInvalidArgs(parser, options, usedOption, array);
+    }
+
+    /**
+     * Check that none of the listed options are present with the combination of used options.
+     */
+    public static void checkInvalidArgsSet(OptionParser parser,
+                                           OptionSet options,
+                                           Set<OptionSpec<?>> usedOptions,
+                                           Set<OptionSpec<?>> invalidOptions,
+                                           Optional<String> trailingAdditionalMessage) {
+        if (usedOptions.stream().filter(options::has).count() == usedOptions.size()) {
+            for (OptionSpec<?> arg : invalidOptions) {
+                if (options.has(arg)) {
+                    printUsageAndDie(parser, String.format("Option combination \"%s\" can't be used with option \"%s\"%s",
+                            usedOptions, arg, trailingAdditionalMessage.orElse("")));
+                }
+            }
+        }
+    }
+
+    public static void printUsageAndDie(OptionParser parser, String message) {
+        System.err.println(message);
+        try {
+            parser.printHelpOn(System.err);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        Exit.exit(1, message);
```

Comment:
Do we need the `Exit.exit(1)` introduced elsewhere in this pull request if this method already uses it internally?

Reply:
Internally? WDYM?

There are some places, where I call this method and then call `Exit.exit(1)` (the Scala version). This is because I need `Nothing` as return value and there is no Java equivalent. It's an ugly hack, but it will naturally go away with the migration of these commands.

Reply:
I had the same question as @clolov , but I guess there's no getting around what you are saying. Either ways, would it make sense to consolidate all the duplicate method calls i.e

```
CommandLineUtils.printUsageAndDie(opts.parser, message)
Exit.exit(1)
```

into a separate method and add a comment to it saying the scala Exit is temporary and not needed when migrating to Java? That should make it easier for the immediate future migrations and reduce the duplicacy. WDYT?

Reply:
Yeah, good idea. Done.

Reply:
Ah, got it. I wasn't aware that Scala needs a different Exit to the one already defined in this method! I agree with the subsequent suggestion.

Reply:
Actually, sorry, I still don't get it and I should have explained my question better. What I meant to ask is whether we can put all calls to Exit.exit(1) only in `CommandLineUtils` functions where we expact a termination of the program and not make explicit reference to it in other commands? For example, here https://github.com/apache/kafka/blob/0803659dc5b5e0aeae82b29e6ba4d07cb855bc1c/core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala#L74 we have pushed Exit.exit(1) down into `CommandLineUtils`. And here https://github.com/apache/kafka/blob/0803659dc5b5e0aeae82b29e6ba4d07cb855bc1c/core/src/main/scala/kafka/admin/ZkSecurityMigrator.scala#L103-L104 we have not.